### PR TITLE
clk2fflogic: nice names for autogenerated signals

### PR DESCRIPTION
**Pull request by nmoroze**
This PR modifies clk2fflogic to use nice, human-readable names for some of the signals it generates when transforming a design. I made this change for a verification project I'm working on where having nice signal names in the SMT2 output comments is helpful, and thought I'd open a PR in case the maintainers are interested in upstreaming this change.

For reference, here's a before-and-after example of the names as they appear in the SMT2 output:

**Before**
```
(declare-datatype |prim_fifo_async_s| ((|prim_fifo_async_mk|
  (|prim_fifo_async_is| Bool)
  (|prim_fifo_async#0| (_ BitVec 16)) ; $auto$clk2fflogic.cc:149:execute$681
  (|prim_fifo_async#1| (_ BitVec 2)) ; $auto$clk2fflogic.cc:146:execute$679
  (|prim_fifo_async#2| (_ BitVec 16)) ; $auto$clk2fflogic.cc:143:execute$677
  (|prim_fifo_async#3| (_ BitVec 1)) ; $auto$clk2fflogic.cc:127:execute$673
  (|prim_fifo_async#4| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$666
  (|prim_fifo_async#5| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$662
  (|prim_fifo_async#6| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$658
  (|prim_fifo_async#7| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$656
  (|prim_fifo_async#8| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$649
  (|prim_fifo_async#9| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$645
  (|prim_fifo_async#10| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$641
  (|prim_fifo_async#11| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$639
  (|prim_fifo_async#12| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$632
  (|prim_fifo_async#13| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$628
  (|prim_fifo_async#14| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$624
  (|prim_fifo_async#15| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$622
  (|prim_fifo_async#16| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$615
  (|prim_fifo_async#17| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$611
  (|prim_fifo_async#18| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$607
  (|prim_fifo_async#19| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$605
  (|prim_fifo_async#20| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$598
  (|prim_fifo_async#21| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$594
  (|prim_fifo_async#22| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$590
  (|prim_fifo_async#23| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$588
  (|prim_fifo_async#24| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$581
  (|prim_fifo_async#25| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$577
  (|prim_fifo_async#26| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$573
  (|prim_fifo_async#27| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$571
  (|prim_fifo_async#28| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$564
  (|prim_fifo_async#29| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$560
  (|prim_fifo_async#30| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$556
  (|prim_fifo_async#31| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$554
  (|prim_fifo_async#32| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$547
  (|prim_fifo_async#33| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$543
  (|prim_fifo_async#34| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$539
  (|prim_fifo_async#35| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$537
  (|prim_fifo_async#36| (_ BitVec 1)) ; $auto$clk2fflogic.cc:42:wrap_async_control$530
  (|prim_fifo_async#37| (_ BitVec 3)) ; $auto$clk2fflogic.cc:215:execute$526
  (|prim_fifo_async#38| (_ BitVec 1)) ; $auto$clk2fflogic.cc:191:execute$522
  (|prim_fifo_async#39| (_ BitVec 3)) ; $auto$clk2fflogic.cc:179:execute$520
  (|prim_fifo_async#40| Bool) ; \clk_rd_i
  (|prim_fifo_async#41| Bool) ; \clk_wr_i
  (|prim_fifo_async#42#final| (Array (_ BitVec 2) (_ BitVec 16))) ; storage
  (|prim_fifo_async#45| (_ BitVec 1)) ; \rst_rd_ni
  (|prim_fifo_async#69| Bool) ; \rready
  (|prim_fifo_async#70| Bool) ; \rst_wr_ni
  (|prim_fifo_async#72| (_ BitVec 16)) ; \wdata
  (|prim_fifo_async#92| Bool) ; \wvalid
  (|prim_fifo_async#42#0| (Array (_ BitVec 2) (_ BitVec 16))) ; storage
)))
```
**After**
```
(declare-datatype |prim_fifo_async_s| ((|prim_fifo_async_mk|
  (|prim_fifo_async_is| Bool)
  (|prim_fifo_async#0| (_ BitVec 16)) ; \storage$0#data_q
  (|prim_fifo_async#1| (_ BitVec 2)) ; \storage$0#addr_q
  (|prim_fifo_async#2| (_ BitVec 16)) ; \storage$0#en_q
  (|prim_fifo_async#3| (_ BitVec 1)) ; \storage$0#past_clk#\clk_wr_i
  (|prim_fifo_async#4| (_ BitVec 1)) ; \fifo_wptr#past_arst#\rst_wr_ni
  (|prim_fifo_async#5| (_ BitVec 3)) ; \fifo_wptr#past_d_wire
  (|prim_fifo_async#6| (_ BitVec 1)) ; \fifo_wptr#past_clk#\clk_wr_i
  (|prim_fifo_async#7| (_ BitVec 3)) ; \fifo_wptr#past_q_wire
  (|prim_fifo_async#8| (_ BitVec 1)) ; \fifo_wptr_gray#past_arst#\rst_wr_ni
  (|prim_fifo_async#9| (_ BitVec 3)) ; \fifo_wptr_gray#past_d_wire
  (|prim_fifo_async#10| (_ BitVec 1)) ; \fifo_wptr_gray#past_clk#\clk_wr_i
  (|prim_fifo_async#11| (_ BitVec 3)) ; \fifo_wptr_gray#past_q_wire
  (|prim_fifo_async#12| (_ BitVec 1)) ; \fifo_rptr#past_arst#\rst_rd_ni
  (|prim_fifo_async#13| (_ BitVec 3)) ; \fifo_rptr#past_d_wire
  (|prim_fifo_async#14| (_ BitVec 1)) ; \fifo_rptr#past_clk#\clk_rd_i
  (|prim_fifo_async#15| (_ BitVec 3)) ; \fifo_rptr#past_q_wire
  (|prim_fifo_async#16| (_ BitVec 1)) ; \fifo_rptr_gray#past_arst#\rst_rd_ni
  (|prim_fifo_async#17| (_ BitVec 3)) ; \fifo_rptr_gray#past_d_wire
  (|prim_fifo_async#18| (_ BitVec 1)) ; \fifo_rptr_gray#past_clk#\clk_rd_i
  (|prim_fifo_async#19| (_ BitVec 3)) ; \fifo_rptr_gray#past_q_wire
  (|prim_fifo_async#20| (_ BitVec 1)) ; \fifo_rptr_sync#past_arst#\rst_wr_ni
  (|prim_fifo_async#21| (_ BitVec 3)) ; \fifo_rptr_sync#past_d_wire
  (|prim_fifo_async#22| (_ BitVec 1)) ; \fifo_rptr_sync#past_clk#\clk_wr_i
  (|prim_fifo_async#23| (_ BitVec 3)) ; \fifo_rptr_sync#past_q_wire
  (|prim_fifo_async#24| (_ BitVec 1)) ; \sync_wptr.intq#past_arst#\rst_rd_ni
  (|prim_fifo_async#25| (_ BitVec 3)) ; \sync_wptr.intq#past_d_wire
  (|prim_fifo_async#26| (_ BitVec 1)) ; \sync_wptr.intq#past_clk#\clk_rd_i
  (|prim_fifo_async#27| (_ BitVec 3)) ; \sync_wptr.intq#past_q_wire
  (|prim_fifo_async#28| (_ BitVec 1)) ; \sync_wptr.q#past_arst#\rst_rd_ni
  (|prim_fifo_async#29| (_ BitVec 3)) ; \sync_wptr.q#past_d_wire
  (|prim_fifo_async#30| (_ BitVec 1)) ; \sync_wptr.q#past_clk#\clk_rd_i
  (|prim_fifo_async#31| (_ BitVec 3)) ; \sync_wptr.q#past_q_wire
  (|prim_fifo_async#32| (_ BitVec 1)) ; \sync_rptr.intq#past_arst#\rst_wr_ni
  (|prim_fifo_async#33| (_ BitVec 3)) ; \sync_rptr.intq#past_d_wire
  (|prim_fifo_async#34| (_ BitVec 1)) ; \sync_rptr.intq#past_clk#\clk_wr_i
  (|prim_fifo_async#35| (_ BitVec 3)) ; \sync_rptr.intq#past_q_wire
  (|prim_fifo_async#36| (_ BitVec 1)) ; \sync_rptr.q#past_arst#\rst_wr_ni
  (|prim_fifo_async#37| (_ BitVec 3)) ; \sync_rptr.q#past_d_wire
  (|prim_fifo_async#38| (_ BitVec 1)) ; \sync_rptr.q#past_clk#\clk_wr_i
  (|prim_fifo_async#39| (_ BitVec 3)) ; \sync_rptr.q#past_q_wire
  (|prim_fifo_async#40| Bool) ; \clk_rd_i
  (|prim_fifo_async#41| Bool) ; \clk_wr_i
  (|prim_fifo_async#42#final| (Array (_ BitVec 2) (_ BitVec 16))) ; storage
  (|prim_fifo_async#45| (_ BitVec 1)) ; \rst_rd_ni
  (|prim_fifo_async#69| Bool) ; \rready
  (|prim_fifo_async#70| Bool) ; \rst_wr_ni
  (|prim_fifo_async#72| (_ BitVec 16)) ; \wdata
  (|prim_fifo_async#92| Bool) ; \wvalid
  (|prim_fifo_async#42#0| (Array (_ BitVec 2) (_ BitVec 16))) ; storage
)))
```

Happy to hear feedback on different ways to do this if you're interested in having nicer names but want the names themselves to look different in some way. 

Thanks!

### CODE REVIEW

File: passes/sat/clk2fflogic.cc [hunk 1]
```diff
@@ -124,7 +127,7 @@ struct Clk2fflogicPass : public Pass {
 								wport, log_id(module), log_id(cell), log_signal(clk),
 								log_signal(addr), log_signal(data));
 
-						Wire *past_clk = module->addWire(NEW_ID);
+						Wire *past_clk = module->addWire(stringf("\\%s$%d#past_clk#%s", log_id(cell), wport, log_signal(clk)));
```
Comment (clairexen) [inline passes/sat/clk2fflogic.cc:130]: This doesn't check if such an identifier already exists. Also, except for very few exceptions, the `\\...` names should only be used for identifiers taken from the original HDL we are processing. Auto-generated names should always be of the `$...` kind.

**Comment by mwkmwkmwk** [comment]
@nmoroze are you still interested in merging this PR? As @clairexen mentioned, it needs naming changes for the generated cells.

**Comment by nmoroze** [comment]
@mwkmwkmwk Thanks for the bump - I should be able to get around to fixing this PR sometime in the next couple of days. If not I'll close it to reduce clutter.

**Comment by nmoroze** [comment]
I've updated the commit to insert the human-readable names as part of the standard autogenerated IDs (including the unique index), so names should now be unique and start with `$`. Please let me know if there are any other changes I should make to the way this is done!

**Review by clairexen — APPROVED** [review]
LGTM

**Pull request merged**